### ISSUE TITLE
Duplicate Tasks or Tasks are getting Skipped

### ISSUE THREAD

**Issue by anandwana001** [google/ground-android#2733]
Looks like there is an issue with the _taskDataFlow in AbstractTaskViewModel

https://github.com/user-attachments/assets/ccd262f7-6051-4c4e-b638-b680a3c02541

**Comment by jabramowitz5** [comment]
@anandwana001 @jo-spek I do not think the behavior in this video is an error, but actually the survey "skipping" over the tasks that should be skipped due to the conditional answer to the "District:" question. The survey is "skipping" over the other potential child questions of the parent conditional. Probably ideally we don't want this to appear like this since it appears like skipping due to conditional and might confuse users, but less of an issue than the actual skipping occurring later in the survey (ie skipping take a photo, capture location, and a few other questions).

**Comment by gino-m** [comment]
This is likely due to the same cause as #2729 and #2727. Patch to `SurveySelectorViewModel` for @sufyanAbbasi:

```kt
  fun activateSurvey(surveyId: String) =
    viewModelScope.launch {
      runCatching {
          // Ignoring extra clicks while loading!
          if (_uiState.value is UiState.ActivatingSurvey) {
            return@runCatching
          }
```

**Comment by gino-m** [comment]
Fixed by #2740. @jabramowitz5 can you please set status to "Verified" if this works once users remove and reopen their surveys without clicking the survey multiple times?
